prov/tcp/src/tcpx_cq.c

```diff
@@ -60,6 +60,39 @@ static int tcpx_cq_close(struct fid *fid)
 	return 0;
 }
 
+static void tcpx_xfer_entry_reset(struct tcpx_buf_pool *pool,
+                                  struct tcpx_xfer_entry *xfer_entry)
+{
+	memset(xfer_entry, 0, sizeof(*xfer_entry));
+
+	xfer_entry->msg_hdr.hdr.version = OFI_CTRL_VERSION;
+	xfer_entry->msg_hdr.hdr.op_data = pool->op_type;
```

Comment:
I realize the old code did this, but it's silly to set op_data here, then overwrite it immediately below in nearly all cases.

Reply:
We can make this memset function more selective regarding the fileds that need to be reset.